index.js

```diff
@@ -546,14 +559,40 @@ class KeyringController extends EventEmitter {
    * @param {string} password - The keyring controller password.
    * @returns {Promise<boolean>} Resolves to true once keyrings are persisted.
    */
-  persistAllKeyrings(password = this.password) {
-    if (typeof password !== 'string') {
+  persistAllKeyrings(password) {
+    if (password && typeof password !== 'string') {
       return Promise.reject(
         new Error('KeyringController - password is not a string'),
       );
     }
 
-    this.password = password;
+    // MV3: Since we also allow persisting without a password, we should require this.encryptedKey
+    if (password === undefined && this.encryptedKey === undefined) {
```

Comment:
All of our tests are failing on this check, which is problematic because `persistAllKeyrings` is called without a password and without encryption key, from a few different places.

Reply:
I believe the problem is that before the changes in this PR, the tests relied on the `this.password = password;` of what used to be line 522 in the `createFirstKeyTree` function

The stack trace for the error in the tests is:
```
  at KeyringController.persistAllKeyrings (index.js:573:9)
      at index.js:239:21
      at Object.<anonymous> (test/index.js:31:5)
```

So it is happening after calling `await keyringController.createNewVaultAndKeychain(password);` on line 31 in the test file, which is in the `beforeEach` function. `createNewVaultAndKeychain` calls `this.createFirstKeyTree(password)`, but passing the password to `createFirstKeyTree` no longer does anything. So...

I think the fix is to pass the password from `createFirstKeyTree` to `this.addNewKeyring` to `this.persistAllKeyrings()`